nca/Resources/NetworkPolicy.py

```diff
@@ -376,10 +377,71 @@ class PolicyConnections:
 class OptimizedPolicyConnections:
     """
     A class to contain the effect of applying policies to all src and dst peers
+    It also serves as a filter for lazy evaluations of connections:
+    whenever any of allowed_conns/denied_conns/pass_conns/all_allowed_conns is None,
+    those connections will not be calculated.
     """
-    def __init__(self):
+    def __init__(self, make_allowed=True, make_denied=True, make_pass=True, make_all_allowed=True):
         self.captured = PeerSet()
```

Comment:
I think it is a bit confusing to use that same type for the new filter as well, this class is now used for two different purposes. It may be more clear with a dedicated type for the filter. 
Also, having a method such as `compute_all_allowed_conns()` can be more clear than comparing some field to `None`.